File: week2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string user_input = get_string("Text: ");

    int letters = count_letters(user_input);
    int words = count_words(user_input);
    int sentences = count_sentences(user_input);


    printf("%i letters\n", letters);
    printf("%i words\n", words);
    printf("%i sentences\n", sentences);

    // 動画で紹介された式を使う。
    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;

    // Coleman-Liau指数は課題で提示された式を使う。
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    // 結果表示
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// 文字数カウント
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

// 単語数カウント
int count_words(string text)
{
    int count = 1; // スペース数 + 1
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

// 文数カウント
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: week6/readability.py
def main():
    user_input = input("Text: ")

    letters = count_letters(user_input)
    words = count_words(user_input)
    sentences = count_sentences(user_input)

    print(f"{letters} letters")
    print(f"{words} words")
    print(f"{sentences} sentences")

    # 動画で紹介された式を使う。
    L = letters / words * 100
    S = sentences / words * 100

    # Coleman-Liau指数は課題で提示された式を使う。
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    # 結果表示
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


# 文字数カウント
def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha(): 
            count += 1
    return count

# 単語数カウント
def count_words(text):
    count = 1  # スペース数 + 1
    for char in text:
        if char == ' ':
            count += 1
    return count



# 文数カウント
def count_sentences(text):
    count = 0
    for char in text:
        if char in ['.', '!', '?']:
            count += 1
    return count

main()
